### PR TITLE
fix wrong params for the server class

### PR DESCRIPTION
**Pull request by khaefeli**
fix for #232

### CODE REVIEW

File: lib/facter/client_crt.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+openvpn_ca = Dir.entries('/etc/openvpn').select {|entry| File.directory? File.join('/etc/openvpn',entry) and !(entry =='.' || entry == '..' || entry == 'keys') }
+
+
+Facter.add(:openvpn_clients) do
+  setcode do
+    openvpn_clients.join(',') if openvpn_clients
+  end
+end
+
+Facter.add(:openvpn_ca) do
```
Comment (ekohl) [inline lib/facter/client_crt.rb]: This and the next 3 facts should handle the case where ```openvpn_ca``` is empty I think.

File: lib/puppet/type/vpnserver.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+module Puppet
+Type.newtype(:vpnserver) do
+  @doc = "Install certificates for the openvpn server slave. 
+  Copied from the master over PuppetDB"
```
Comment (ekohl) [inline lib/puppet/type/vpnserver.rb]: Isn't this an provider detail rather than the type?

File: manifests/client.pp [hunk 3]
```diff
@@ -218,7 +218,9 @@
   $pull                 = false,
   $server_extca_enabled = false
 ) {
-
+  err ($::openvpn_clients)
```
Comment (ekohl) [inline manifests/client.pp]: This looks like test code.

File: manifests/slave.pp [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+define openvpn::slave(
+) {
+  include openvpn
+  Class['openvpn::install'] ->
+  Openvpn::Slave[$name]
+
+  Vpnserver <<| |>>
```
Comment (ekohl) [inline manifests/slave.pp]: I can imagine that some users might have multiple sets of VPN servers so you might not want to collect everything.

File: lib/facter/client_crt.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
+require 'facter'
+
+if File.directory?('/etc/openvpn/')
+openvpn_clients = Dir['/etc/openvpn/*/easy-rsa/keys/*.crt'].map { |a| a.gsub(%r{\.crt$}, '').gsub(%r{^.*/}, '') }
```
[the remark this answers is not in the document]
Comment (khaefeli) [reply lib/facter/client_crt.rb]: @ekohl thanks for reviewing. 
Sorry, I've created a PR from my master by mistake. Should be another branch for the params fix. (without my facter tests ;) )

**Pull request closed without merging**